doc/designs/fido2.md

```diff
@@ -0,0 +1,239 @@
+
+Note: since the Web browser my be running on a non-enrolled host without
+the required packages, the WebUI will probably need specific javascript code
+to register a key by inserting it on the machine where the browser is
+running.
+
+Investigations TBD regarding the possible solutions.
+
+
+### CLI
+
+| Command |	Options | Description |
+| --- | ----- | --- |
+| | **FIDO2 configuration** |
+| fido2config-show | | This command displays the FIDO2 settings |
+| fido2config-mod | --require-user-verif=['on', 'off', 'default'] | This command modifies the FIDO2 settings |
+| | **User Mapping** | |
+| user-add-fido2key | LOGIN [FIDO2KEY...] | This command does not require the FIDO2 key to be inserted and can directly add the mapping data, obtained through another mean (for instance through fido2_child) |
+| user-add-fido2key | LOGIN --register [--cose-type=['es256', 'rs256', 'eddsa']] [--require-user-verif] | This command requires the insertion of the FIDO2 key, performs the registration with the specified cose type + user verification requirement, and adds the mapping data to the user entry |
+| user-remove-fido2key | LOGIN FIDO2KEY... | |
+| user-show | LOGIN | This command displays the FIDO2 key mapping if set, with the label `FIDO2 key mapping` |
+| stageuser-add-fido2key | LOGIN [FIDO2KEY...] | This command does not require the FIDO2 key to be inserted and can directly add the mapping data, obtained through another mean (for instance through fido2_child) |
+| stageuser-add-fido2key | LOGIN --register [--cose-type=['es256', 'rs256', 'eddsa']] [--require-user-verif] | This command requires the insertion of the FIDO2 key, performs the registration with the specified cose type + user verification requirement, and adds the mapping data to the user entry |
+| stageuser-remove-fido2key | LOGIN FIDO2KEY... | |
+| stageuser-show | LOGIN | This command displays the FIDO2 key mapping if set, with the label `FIDO2 key mapping` |
+
+
+### Configuration
+
+The global settings can be read or modified using `ipa fido2config-[show|mod]`.
+
+## Upgrade
+
+During upgrade, the new LDAP schema is automatically added and replicated to the replicas.
+
+The upgrade must create the FIDO2 configuration entry if it does not already exist, with value='default' for the 'require user verification' setting (meaning it will use the default value from each key).
+
+## Test plan
+
+XMLRPC tests must validate the new CLI.
+
+## Troubleshooting and debugging
+
+SSSD's helper `fido2_child` provides debugging options:
+
+`fido2_child --register --username=fido2user --domain=ipa.test --debug-level=9 --logger=stderr --debug-libfido2`
+
+SSSD's helper can also be used to test the authentication:
+
+`fido_child --authenticate --username=fido2user --domain=ipa.test --public-key=... --key-handle=... --debug-level=9 --logger=stderr --debug-libfido2`
```

Comment:
s/fido2_child/sssctl fido2

Reply:
It's sssctl fido2-exec

Reply:
Mentioned the 2 commands in the troubleshooting section.